internal/pkg/agent/cmd/container_init_test.go

```diff
@@ -0,0 +1,203 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+//go:build linux
+
+package cmd
+
+import (
+	"os"
+	"path/filepath"
+	"syscall"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"kernel.org/pub/linux/libs/security/libcap/cap"
+)
+
+func Test_chownPaths(t *testing.T) {
+	firstParentDir, err := os.MkdirTemp("", "test_chown")
+	require.NoError(t, err)
+	defer os.RemoveAll(firstParentDir)
+
+	secondParentDir, err := os.MkdirTemp("", "test_chown")
+	require.NoError(t, err)
+	defer os.RemoveAll(secondParentDir)
```

Comment:
[Suggestion]
You could use `t.TempDir()` instead, then Go will take care of the cleanup.